virtcontainers/pkg/oci/utils.go

```diff
@@ -222,7 +222,81 @@ func newLinuxDeviceInfo(d spec.LinuxDevice) (*config.DeviceInfo, error) {
 	return &deviceInfo, nil
 }
 
-func containerDeviceInfos(spec CompatOCISpec) ([]config.DeviceInfo, error) {
+func configContainsPCIColdplugGroup(hConfig *vc.HypervisorConfig, iommuGroup string) bool {
+	for _, g := range hConfig.PCIColdplugGroups {
+		if iommuGroup == strings.TrimSpace(g) {
+			return true
+		}
+	}
+	return false
+}
+
+// getBDF returns the BDF of pci device
+// Expected input strng format is [<domain>]:[<bus>][<slot>].[<func>] eg. 0000:02:10.0
+func getBDF(deviceSysStr string) (string, error) {
+	tokens := strings.Split(deviceSysStr, ":")
+
+	if len(tokens) != 3 {
+		return "", fmt.Errorf("Incorrect number of tokens found while parsing bdf for device : %s", deviceSysStr)
+	}
+
+	tokens = strings.SplitN(deviceSysStr, ":", 2)
+	return tokens[1], nil
+}
+
+func getBDFsFromIOMMU(iommuGroup string) ([]string, error) {
+	var bdfs []string
+	iommuDevicesPath := filepath.Join(config.SysIOMMUPath, iommuGroup, "devices")
+
+	deviceFiles, err := ioutil.ReadDir(iommuDevicesPath)
+	if err != nil {
+		return nil, err
+	}
+
+	for _, deviceFile := range deviceFiles {
+		deviceBDF, err := getBDF(deviceFile.Name())
+		if err != nil {
+			return nil, err
+		}
+		bdfs = append(bdfs, deviceBDF)
+	}
+
+	return bdfs, nil
+}
+
+func checkForPCIColdPlug(device spec.LinuxDevice, hConfig *vc.HypervisorConfig) bool {
+	if !strings.Contains(device.Path, "/dev/vfio/") {
+		return false
+	}
+
+	pathTokens := strings.Split(device.Path, "/")
+	if len(pathTokens) != 4 {
```

Comment:
Could you add a comment documenting the expected format so this token value is clearer?